testsuite/tests/kuadrant/authorino/identity/rhsso/test_jwt_ttl.py

```diff
@@ -0,0 +1,66 @@
+"""
+Test for JWT TTL (time to live), see
+https://github.com/Kuadrant/authorino/blob/main/docs/features.md#jwt-verification-authenticationjwt
+"""
+
+from time import sleep
+
+import pytest
+
+pytestmark = [pytest.mark.authorino]
+
+
+def test_jwt_ttl(client, auth, rhsso, create_jwt, jwt_ttl):
+    """
+    Note that similar test exists also in test_dinosaur.py. If modifying this test update the test there too.
+
+    Test:
+        - send request using user with valid middle name
+        - assert that response status code is 200
+        - delete the signing RS256 key from JWKS - the one associated with the JWT for user with valid middle name
+        - assert that response status code is still 200 due to deleted key being cached by Authorino
+        - sleep to allow for Authorino to update the cache as part of OIDC discovery
+        - assert that response status is 401 after cache update
+        - create a new signing RS256 key in JWKS to be used for generating new JWTs
+        - generate a new JWT
+        - assert that response status code is 200 if using new JWT
+        - assert that response status is still 401 for old JWT - ie old JWT does not work with new signing RS256 key
+
+    """
+    response = client.get("/anything/dinosaurs_mgmt/v1/dinosaurs", auth=auth)
+    assert response.status_code == 200
+
+    # delete the current signing RS256 JWKS key
+    rhsso.delete_signing_rs256_jwks_key()
+
+    # 200 OK expected since Authorino should have the deleted JWKS key still cached.
+    # Potentially unstable if Authorino triggers OIDC discovery in between the JWKS key deletion just above
+    # and the client.get call just below.
+    response = client.get("/anything/dinosaurs_mgmt/v1/dinosaurs", auth=auth)
+    assert response.status_code == 200
+
+    # Sleeping for jwt_ttl seconds (max 60s though) to ensure Authorino triggers the OIDC discovery during the sleep.
+    # OIDC discovery detects that the signing RS256 JWKS key has been removed and updates the Authorino cache.
+    sleep(jwt_ttl if jwt_ttl <= 60 else 30)
```

Comment:
in what scenario is jwt_ttl higher than 60sec?

Reply:
Good question. The answer is that there is no such scenario, it is just a safety measure in case somebody changes the value for some long interval for whatever reason (e.g. testing purposes). If you don't like it it is perfectly ok to remove it but in that case I would add a comment to `jwt_ttl` fixture saying that this should not be set for too many seconds. Wdyt?

Reply:
Why is there a maximum of 60sec?

Reply:
This is apparently confusing so I will remove this safety measure and just add a comment to the fixture that if the value is modified to higher number then some test might take long to finish